packages/dev/core/src/Culling/Helper/boundingInfoHelper.ts

```diff
@@ -1,210 +1,54 @@
-            }
-
-            // Morph
-            const manager = mesh ? (<Mesh>mesh).morphTargetManager : null;
-            if (manager) {
-                numInfluencers = manager.numMaxInfluencers || manager.numInfluencers;
-                if (numInfluencers > 0) {
-                    defines.push("#define MORPHTARGETS");
-                }
-                if (manager.isUsingTextureForTargets) {
-                    defines.push("#define MORPHTARGETS_TEXTURE");
-
-                    if (uniforms.indexOf("morphTargetTextureIndices") === -1) {
-                        uniforms.push("morphTargetTextureIndices");
-                    }
-
-                    if (samplers.indexOf("morphTargets") === -1) {
-                        samplers.push("morphTargets");
-                    }
-                }
-                defines.push("#define NUM_MORPH_INFLUENCERS " + numInfluencers);
-                for (let index = 0; index < numInfluencers; index++) {
-                    attribs.push(VertexBuffer.PositionKind + index);
-                }
-                if (numInfluencers > 0) {
-                    uniforms = uniforms.slice();
-                    uniforms.push("morphTargetInfluences");
-                    uniforms.push("morphTargetCount");
-                    uniforms.push("morphTargetTextureInfo");
-                    uniforms.push("morphTargetTextureIndices");
-                }
-            }
-
-            // Baked Vertex Animation
-            const bvaManager = (<Mesh>mesh).bakedVertexAnimationManager;
-
-            if (bvaManager && bvaManager.isEnabled) {
-                defines.push("#define BAKED_VERTEX_ANIMATION_TEXTURE");
-                if (uniforms.indexOf("bakedVertexAnimationSettings") === -1) {
-                    uniforms.push("bakedVertexAnimationSettings");
-                }
-                if (uniforms.indexOf("bakedVertexAnimationTextureSizeInverted") === -1) {
-                    uniforms.push("bakedVertexAnimationTextureSizeInverted");
-                }
-                if (uniforms.indexOf("bakedVertexAnimationTime") === -1) {
-                    uniforms.push("bakedVertexAnimationTime");
-                }
-
-                if (samplers.indexOf("bakedVertexAnimationTexture") === -1) {
-                    samplers.push("bakedVertexAnimationTexture");
-                }
-                PrepareAttributesForBakedVertexAnimation(attribs, mesh, defines);
-            }
-
-            const join = defines.join("\n");
-            if (!this._effects[join]) {
-                computeEffect = this._platform.createUpdateEffect(attribs, join, uniforms, samplers, numInfluencers);
-                this._effects[join] = computeEffect;
+    public async computeAsync(target: AbstractMesh | AbstractMesh[]): Promise<void> {
+        if (!this._platform) {
+            if (this._engine.getCaps().supportComputeShaders) {
+                const module = await import("./computeShaderBoundingHelper");

```

Comment:
Nice! @RaananW what does this do for our UMD bundle? I assume this just gets included in the full bundle and wrapped into a Promise returning function?

Reply:
Yup it is through its presence in the index file otherwise it would appear in its own chunck.